contracts/Manager.sol

```diff
@@ -333,364 +214,190 @@ contract Manager is UUPSOwnableUpgradeable, UsingRegistryUpgradeable {
-     * their voting limit, causing a larger deposit to revert with
-     * NoVotableGroups, despite there still being some room for deposits, this
-     * can be worked around by sending a few smaller deposits.
-     */
-    function distributeVotes(uint256 votes) internal {
-        /*
-         * "Votable" groups are those that will currently fit under the voting
-         * limit in Election.sol even if voted for with the entire `votes`
-         * amount. Note that some might still not end up getting voted for given
-         * the distribution logic below.
-         */
-        address[] memory votableGroups = getVotableGroups(votes);
-        if (votableGroups.length == 0) {
-            revert NoVotableGroups();
-        }
-
-        GroupWithVotes[] memory sortedGroups;
-        uint256 availableVotes;
-        (sortedGroups, availableVotes) = getSortedGroupsWithVotes(votableGroups);
-        availableVotes += votes;
-
-        uint256[] memory votesPerGroup = new uint256[](votableGroups.length);
-        uint256 groupsVoted = votableGroups.length;
-        uint256 targetVotes = availableVotes / groupsVoted;
-
-        /*
-         * This would normally be (i = votableGroups.length - 1; i >=0; i--),
-         * but we can't i-- on the last iteration when i=0, since i is an
-         * unsigned integer. So we iterate with the loop variable 1 greater than
-         * expected, set index = i-1, and use index inside the loop.
-         */
-        for (uint256 i = votableGroups.length; i > 0; i--) {
-            uint256 index = i - 1;
-            if (sortedGroups[index].votes >= targetVotes) {
-                groupsVoted--;
-                availableVotes -= sortedGroups[index].votes;
-                targetVotes = availableVotes / groupsVoted;
-                votesPerGroup[index] = 0;
-            } else {
-                votesPerGroup[index] = targetVotes - sortedGroups[index].votes;
-
-                if (availableVotes % groupsVoted > index) {
-                    votesPerGroup[index]++;
+     * @notice Used to deposit CELO into the StakedCelo system. The user will
+     * receive an amount of stCELO proportional to their contribution. The CELO
+     * will be scheduled to be voted for with the Account contract.
+     * The CELO will be distributed based on accoutn strategy.
+     */
+    function deposit() external payable {
+        address strategy = _checkAndUpdateStrategy(msg.sender, strategies[msg.sender]);
+
+        uint256 stCeloValue = toStakedCelo(msg.value);
+        if (strategy != address(0)) {
+            if (!groupHealth.isValidGroup(strategy)) {
+                // if invalid group vote for default strategy
+                strategies[msg.sender] = address(0);
+                strategy = address(0);
+                uint256 stCeloBalance = stakedCelo.balanceOf(msg.sender);
+                if (stCeloBalance != 0) {
+                    _transfer(strategy, address(0), stCeloBalance, msg.sender, msg.sender);
```

Comment:
Shouldn't we still set the strategy to the one specified by the caller, so that votes will be transferred to that group if it later becomes eligible for the specific group strategy?

Reply:
This will be handled as part of overflow task as discussed offline